pkg/diagnostics/grpc_tracing.go

```diff
@@ -33,15 +34,47 @@ import (
 )
 
 const (
-	GRPCTraceContextKey       = "grpc-trace-bin"
-	GRPCProxyAppIDKey         = "dapr-app-id"
 	daprInternalPrefix        = "/dapr.proto.internals."
 	daprRuntimePrefix         = "/dapr.proto.runtime."
 	daprInvokeServiceMethod   = "/dapr.proto.runtime.v1.Dapr/InvokeService"
 	daprCallLocalStreamMethod = "/dapr.proto.internals.v1.ServiceInvocation/CallLocalStream"
 	daprWorkflowPrefix        = "/TaskHubSidecarService"
 )
 
+// handleBaggage extracts baggage from the incoming metadata and forwards that along as metadata,
+// and checks for context baggage via otel context checking and propagates that along in the ctx.
+// There are 2 separate streams in which baggage can come in and flow out and each is respected
+// and NOT combined for security reasons.
+func handleBaggage(ctx context.Context) (context.Context, string) {
+	// metadata baggage (do not inject into context)
+	md, _ := grpcMetadata.FromIncomingContext(ctx)
+	if md == nil {
+		md = make(grpcMetadata.MD)
```

Comment:
Should this not be an early return?

Reply:
This indicates that there may not be header baggage, but we still need to check for the ctx baggage and if that is valid by otel's standards, so no we should not return early here.